File: node-http/models/groupRelation.js
/**
 * @Author:acexy@thankjava.com
 * 18/8/20
 * @Description:
 */
const baseModel = require('./basic.js');

class groupRelation extends baseModel {

    Schema() {
        return {
            groupId: {type: String, required: true},
            userId: {type: String, required: true},
            remark: {type: String, required: false},
        };
    };

    SchemaName() {
        return 'group_relation';
    };

    save(groupRelation) {
        return new this.model(groupRelation).save();
    }

    selectByUserId(userId) {
        return this.model.find({userId: userId}).exec();
    }

    selectByGroupId(groupId) {
        return this.model.find({groupId: groupId}).exec();
    }

}

module.exports = new groupRelation();

File: java-websocket/src/main/java/com/thankjava/wchat/db/entity/GroupRelation.java
package com.thankjava.wchat.db.entity;

/**
 * @Author: acexy@thankjava.com
 * 2018/8/14
 * @Description:
 **/
public class GroupRelation extends BaseEntity {

    private String groupId;
    private String userId;
    private String remark;

    public String getGroupId() {
        return groupId;
    }

    public void setGroupId(String groupId) {
        this.groupId = groupId;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }
}
